### PR TITLE
Postgres plugin integration tests

### PR DESCRIPTION
**Pull request by lijunwangs**
#### Problem
It has been found that the postgres plugin has run into regressions a few times due to changes to Rust models without corresponding changes in the Db side. Integration tests are needed to guard against regressions.

#### Summary of Changes
An integration test has been developed to use the postgres plugin for a locally run cluster. The test covers streaming accounts, transactions, slot and block metadata.

Fixes #

### CODE REVIEW

File: ci/docker-rust/build.sh [hunk 1]
```diff
@@ -8,5 +8,5 @@ docker build -t solanalabs/rust .
 read -r rustc version _ < <(docker run solanalabs/rust rustc --version)
 [[ $rustc = rustc ]]
 docker tag solanalabs/rust:latest solanalabs/rust:"$version"
-docker push solanalabs/rust:"$version"
-docker push solanalabs/rust:latest
+# docker push solanalabs/rust:"$version"
```
Comment (mvines) [inline ci/docker-rust/build.sh:11]: Can you please just remove these from the PR.  I can then pull the PR locally and rebuild the docker image for you

File: ci/test-stable.sh [hunk 2]
```diff
@@ -29,6 +29,9 @@ NPROC=$((NPROC>14 ? 14 : NPROC))
 echo "Executing $testName"
 case $testName in
 test-stable)
+  # start PostgreSQL database
+  /etc/init.d/postgresql start
```
Comment (mvines) [inline ci/test-stable.sh:33]: It's not safe to assume that the host machine will have postgresql installed.  For example it's common to run `ci/test-stable.sh` from your development machine.  The postgresql-dependent tests should just be ignored in this case

Comment (lijunwangs) [reply ci/test-stable.sh:33]: Okay -- we have to check and ignore postgres setup failures and make the test_postgres_plugin to tolerate DB not setup.

Comment (mvines) [reply ci/test-stable.sh:33]: Is there a handy existing postgres env var we could use to indicate the presence of postgres maybe?

File: ci/docker-rust/build.sh (hunk 1 above)
Comment (lijunwangs) [reply ci/docker-rust/build.sh]: Reverted this change.

**Comment by mvines** [comment]
hey @lijunwangs, I just pushed a new `solanalabs/rust:1.57.0` docker image with your changes here

**Comment by lijunwangs** [comment]
> hey @lijunwangs, I just pushed a new `solanalabs/rust:1.57.0` docker image with your changes here

Thanks @mvines  -- I am redoing the build to test against the new docker image to verify.

**Comment by lijunwangs** [comment]
It turns out the postgres integration test does not work well with the current CI environment of using the current host user. PostgreSQL requires the user to be either root or 'postgres' to run the service and it does not tolerate the 'unknown' user. @mvines and I talked about using two containers to workaround and determined it was risky for the stability of the CI system  when debugging and issues spread in two containers. We will use this as the wedge point to move the Postgres plugin out to its own repo and look into using GitHub actions for the integration. Close this PR for now.

**Comment by t-nelson** [comment]
Another great reason to move the Postgres plugin to its own repo! :wink:

**Pull request closed without merging**